unison-share-api/src/Unison/Server/NameSearch/Sqlite.hs

```diff
@@ -0,0 +1,194 @@
+module Unison.Server.NameSearch.Sqlite
+  ( resolveShortHash,
+    typeReferencesByShortHash,
+    termReferentsByShortHash,
+    NameSearch (..),
+    scopedNameSearch,
+  )
+where
+
+import Control.Lens
+import qualified Data.Set as Set
+import U.Codebase.HashTags (BranchHash)
+import qualified U.Codebase.Sqlite.NamedRef as NamedRef
+import qualified U.Codebase.Sqlite.Operations as Ops
+import qualified Unison.Builtin as Builtin
+import Unison.Codebase (Codebase)
+import qualified Unison.Codebase as Codebase
+import Unison.Codebase.Path
+import qualified Unison.Codebase.Path as Path
+import qualified Unison.Codebase.SqliteCodebase.Conversions as Cv
+import qualified Unison.Debug as Debug
+import qualified Unison.HashQualified' as HQ'
+import qualified Unison.LabeledDependency as LD
+import Unison.Name (Name)
+import qualified Unison.Name as Name
+import Unison.NameSegment (NameSegment (..))
+import Unison.Prelude
+import Unison.Reference (Reference)
+import qualified Unison.Reference as Reference
+import Unison.Referent (Referent)
+import qualified Unison.Referent as Referent
+import Unison.Server.NameSearch (NameSearch (..), Search (..))
+import qualified Unison.Server.SearchResult as SR
+import qualified Unison.ShortHash as SH
+import qualified Unison.Sqlite as Sqlite
+import qualified Unison.Util.Set as Set
+
+data SearchStrategy
+  = ExactMatch
+  | SuffixMatch
+  deriving (Show, Eq)
+
+scopedNameSearch :: Codebase m v a -> BranchHash -> Path -> NameSearch Sqlite.Transaction
```

Comment:
Implements the `NameSearch` interface entirely using SQLite so we don't need to load names objects at all 💪🏼